code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm

```diff
@@ -122,6 +122,10 @@
 				OpenFire()
 		else
 			devour(L)
+			return
+
+		if(L.health <= HEALTH_THRESHOLD_DEAD && HAS_TRAIT(L, TRAIT_NODEATH)) //Nope, it still gibs yall
```

Comment:
turn this into an else if between the if and else

Reply:
yeah